homeassistant/components/ps4/config_flow.py

```diff
@@ -95,35 +130,51 @@ def __init__(self):
             self.name = user_input[CONF_NAME]
             self.pin = user_input[CONF_CODE]
             self.host = user_input[CONF_IP_ADDRESS]
-
-            is_ready, is_login = await self.hass.async_add_executor_job(
-                self.helper.link, self.host, self.creds, self.pin)
-
-            if is_ready is False:
-                errors['base'] = 'not_ready'
-            elif is_login is False:
-                errors['base'] = 'login_failed'
-            else:
-                device = {
-                    CONF_HOST: self.host,
-                    CONF_NAME: self.name,
-                    CONF_REGION: self.region
-                }
-
-                # Create entry.
-                return self.async_create_entry(
-                    title='PlayStation 4',
-                    data={
-                        CONF_TOKEN: self.creds,
-                        'devices': [device],
-                    },
-                )
+            pin_valid = True
+
+            # Remove any whitespace from pin.
+            if ' ' in self.pin:
+                self.pin = self.pin.replace(' ', '')
```

Comment:
Do we really need to remove whitespace inside the string? This is the only operation of the pin checks that can't be solved with a voluptuous schema. I suggest simplifying this part and replace all of these checks with a voluptuous schema.

```py
vol.All(
    vol.Strip, vol.Length(min=8, max=8),
    vol.Coerce(int))
```
If there is whitespace inside the string the integer coercion will fail, so we will catch that.

Reply:
Yeah that works better. Except for if wrong length the error shows "User input malformed." Is there a way to change the message as it is unclear as to what is wrong?

Reply:
That should be fixed in the data entry helper or in the frontend, wherever that is handled. So it's a different PR.